src/Magento/FunctionalTestingFramework/StaticCheck/TestDependencyCheck.php

```diff
@@ -0,0 +1,385 @@
+            $modulePath = dirname(dirname(dirname(dirname($filePath))));
+            $moduleFullName = array_search($modulePath, $this->moduleNameToPath) ?? null;
+            // Not a module, is either dev/tests/acceptance or loose folder with test materials
+            if ($moduleFullName == null) {
+                continue;
+            }
+
+            $contents = file_get_contents($filePath);
+            $allEntities = [];
+            preg_match_all(ActionObject::ACTION_ATTRIBUTE_VARIABLE_REGEX_PATTERN, $contents, $braceReferences);
+            preg_match_all(self::ACTIONGROUP_REGEX_PATTERN, $contents, $actionGroupReferences);
+            preg_match_all(self::EXTENDS_REGEX_PATTERN, $contents, $extendReferences);
+
+            // Remove Duplicates
+            $braceReferences[0] = array_unique($braceReferences[0]);
+            $actionGroupReferences[1] = array_unique($actionGroupReferences[1]);
+            $braceReferences[1] = array_unique($braceReferences[1]);
+            $braceReferences[2] = array_filter(array_unique($braceReferences[2]));
+
+            // Check `data` entities in {{data.field}} or {{data.field('param')}}
+            foreach ($braceReferences[0] as $reference) {
+                // trim `{{data.field}}` to `data`
+                preg_match('/{{([^.]+)/', $reference, $entityName);
+                // Double check that {{data.field}} isn't an argument for an ActionGroup
+                $entity = $this->findEntity($entityName[1]);
+                preg_match_all(self::ACTIONGROUP_ARGUMENT_REGEX_PATTERN, $contents, $possibleArgument);
+                if (array_search($entityName[1], $possibleArgument[1]) !== false) {
+                    continue;
+                }
+                if ($entity !== null) {
+                    $allEntities[$entity->getName()] = $entity;
+                }
+            }
+
+            // Drill down into params in {{ref.params('string', $data.key$, entity.reference)}}
+            foreach ($braceReferences[2] as $parameterizedReference) {
+                preg_match(
+                    ActionObject::ACTION_ATTRIBUTE_VARIABLE_REGEX_PARAMETER,
+                    $parameterizedReference,
+                    $arguments
+                );
+                $splitArguments = explode(',', ltrim(rtrim($arguments[0], ")"), "("));
+                foreach ($splitArguments as $argument) {
+                    // Do nothing for 'string' or $persisted.data$
+                    if (preg_match(ActionObject::STRING_PARAMETER_REGEX, $argument)) {
+                        continue;
+                    } elseif (preg_match(TestGenerator::PERSISTED_OBJECT_NOTATION_REGEX, $argument)) {
+                        continue;
+                    }
+                    // trim `data.field` to `data`
+                    preg_match('/([^.]+)/', $argument, $entityName);
+                    // Double check that {{data.field}} isn't an argument for an ActionGroup
```

Comment:
Why skipping? What if it's a direct entity reference in an argument for an ActionGroup? e.g. {{_defatultCategory.name}}

Reply:
Skipping done on line 175 refers to the following case:
```
<argument name="product" defaultValue="_defaultProduct"/>
...
<comment userInput="{{product.name}} />
<comment userInput="{{_defaultProduct.name}} />
```

The entity `product` is just the argument, and if we have a match of `name` we shouldn't try to resolve it, as the entity doesn't exist.

So in this case the first line would be skipped, but line two would not be.

Reply:
ok. This is right. I did not look the regex and just thought the argument you tried to skip is in the "value" part.